### PR TITLE
Added cluster api

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/apis/cluster/v1alpha1/cluster.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kuratorclusters,shortName=kcluster
+// +kubebuilder:subresource:status
+
+// KuratorCluster is the schema for existing node based Kubernetes Cluster API.
+type KuratorCluster struct {
```
Comment (zirain) [inline pkg/apis/cluster/v1alpha1/cluster.go:31]: prefer to user `Cluster` directly

Comment (hzxuzhonghu) [reply pkg/apis/cluster/v1alpha1/cluster.go:31]: The only concern is if we use cluster, it will be same as cluster-api cluster

Comment (zirain) [reply pkg/apis/cluster/v1alpha1/cluster.go]: `CustomCluster` is better.

**Comment by hzxuzhonghu** [comment]
@Xieql Please based on this to write the operator framework

**Pull request merged**